server/src/main/java/org/opensearch/index/store/Store.java

```diff
@@ -1012,6 +1028,53 @@ public Set<String> getPendingDeletions() throws IOException {
             // to be removed once fixed in FilterDirectory.
             return unwrap(this).getPendingDeletions();
         }
+
+        @Override
+        public void copyFrom(Directory from, String src, String dest, IOContext context) throws IOException {
+            long fileSize = from.fileLength(src);
+            beforeDownload(fileSize);
+            boolean success = false;
+            try {
+                long startTime = System.currentTimeMillis();
+                super.copyFrom(from, src, dest, context);
+                success = true;
+                afterDownload(fileSize, startTime);
+            } finally {
+                if (!success) {
+                    downloadFailed(fileSize);
+                }
+            }
+        }
+
+        /**
+         * Updates the amount of bytes attempted for download
+         */
+        private void beforeDownload(long fileSize) {
+            directoryFileTransferTracker.addDownloadBytesStarted(fileSize);
+        }
+
+        /**
+         * Updates
+         * - The amount of bytes that has been successfully downloaded from the source store
+         * - The last successful download completion timestamp
+         * - The last successfully downloaded file
+         * - Download speed (in bytes/sec)
+         */
+        private void afterDownload(long fileSize, long startTimeInMs) {
+            directoryFileTransferTracker.addDownloadBytesSucceeded(fileSize);
+            directoryFileTransferTracker.updateLastDownloadedSegmentSize(fileSize);
```

Comment:
Do we really need to call `updateLastDownloadedSegmentSize` separately? Tracker should be able to call this internally as part of `addDownloadBytesSucceeded`.